pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go

```diff
@@ -338,15 +340,102 @@ func (pl *VolumeRestrictions) EventsToRegister() []framework.ClusterEventWithHin
+		return framework.QueueAfterBackoff
+	}
+
+	return framework.QueueImmediately
+}
+
+// isSchedulableAfterNodeAdded is invoked whenever a node added, It checks whether
+// that change made a previously unschedulable pod schedulable.
+func (pl *VolumeRestrictions) isSchedulableAfterNodeAdded(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) framework.QueueingHint {
+	_, newNode, err := util.As[*v1.Node](oldObj, newObj)
+	if err != nil {
+		logger.Error(err, "unexpected objects in isSchedulableAfterNodeAdded", "oldObj", oldObj, "newObj", newObj)
+		return framework.QueueAfterBackoff
+	}
+
+	logger.V(4).Info("node was added", "pod", klog.KObj(pod), "node", klog.KObj(newNode))
+	return framework.QueueAfterBackoff
+}
+
+// isSchedulableAfterPersistentVolumeClaimChange is invoked whenever a PersistentVolumeClaim changed, It checks whether
+// that change made a previously unschedulable pod schedulable.
+func (pl *VolumeRestrictions) isSchedulableAfterPersistentVolumeClaimChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) framework.QueueingHint {
+	_, modifiedPersistentVolumeClaim, err := util.As[*v1.PersistentVolumeClaim](oldObj, newObj)
+	if err != nil {
+		logger.Error(err, "unexpected objects in isSchedulableAfterPersistentVolumeClaimChange", "oldObj", oldObj, "newObj", newObj)
+		return framework.QueueAfterBackoff
+	}
+
+	usePVC := false
+	// Check if the pod's pvc is created
+	if pl.FilterPVCFromPod(pod, modifiedPersistentVolumeClaim) {
+		usePVC = true
+	}
+
+	if !usePVC {
+		// This was not the PersistentVolumeClaim the pod was waiting for.
+		logger.V(6).Info("unrelated PersistentVolumeClaim got modified", "pod", klog.KObj(pod), "PersistentVolumeClaim", klog.KObj(modifiedPersistentVolumeClaim))
+		return framework.QueueSkip
+	}
+
+	if !pl.enableReadWriteOncePod {
+		logger.V(7).Info("spec of PersistentVolumeClaim for pod got updated", "pod", klog.KObj(pod), "PersistentVolumeClaim", klog.KObj(modifiedPersistentVolumeClaim))
+		return framework.QueueImmediately
+	}
+
+	key := framework.GetNamespacedName(pod.Namespace, modifiedPersistentVolumeClaim.Name)
+	if pl.sharedLister.StorageInfos().IsPVCUsedByPods(key) && v1helper.ContainsAccessMode(modifiedPersistentVolumeClaim.Spec.AccessModes, v1.ReadWriteOncePod) {
+		logger.V(4).Info("Pod uses PVCs with the ReadWriteOncePod access mode", "pod", klog.KObj(pod), "PersistentVolumeClaim", klog.KObj(modifiedPersistentVolumeClaim))
+		return framework.QueueAfterBackoff
+	}
+
+	logger.V(4).Info("spec of PersistentVolumeClaim for pod got updated", "pod", klog.KObj(pod), "PersistentVolumeClaim", klog.KObj(modifiedPersistentVolumeClaim))
+	return framework.QueueImmediately
+}
+
+// FilterPVCFromPod Get the PVC bound to the volume in the pod and compare it to the name and namespace of the parameter pvc
+// return true if the name and namespace are equal, false if not
+func (pl *VolumeRestrictions) FilterPVCFromPod(pod *v1.Pod, pvc *v1.PersistentVolumeClaim) bool {
```

Comment:
You can reuse the `readWriteOncePodPVCsForPod` and remove this function. 
Also I think we should check the pod's RWOP pvc in each `ClusterEventWithHint` first🤔️. For example, if a pod's RWOP pvc is already occupied, then even a new node added won't help.

Reply:
1. `readWriteOncePodPVCsForPod` doesn't help me verify that the new (modified) pvc was created by the current pod, It will filter out all PVCs where access mode is ReadWriteOncePod, if I can't get new (modified) pvc from pod's volume.PersistentVolumeClaim, can we assume that this pvc is not related to the pod and should I return Skip or Immediately at this point? what do you think？

2. Based on what was said in the [comment](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/framework/types.go#L109-L121), I don't think we need to check for pvc in all `ClusterEventWithHint`, because if PVC causes the scheduling to fail, it will retrying the scheduling again after backoff and won't complete the scheduling in this scheduling cycle